src/ObjectStore/v1/Service.php

```diff
@@ -86,4 +86,42 @@ public function containerExists(string $name): bool
             throw $e;
         }
     }
+
+    /**
+     * Creates a temporary URL to access object in private containers.
+     * This method loosely follows swift command's way to generate temporary url: `swift tempurl $METHOD $EXPIRE $PATH $KEY`.
+     *
+     * @param string $method  An HTTP method to allow for this temporary URL. Any of GET, POST, HEAD, PUT, POST, DELETE.
+     * @param int    $expires Unix timestamp
+     * @param string $path    The full path or storage URL to the Swift object. Example: '/v1/AUTH_account/c/o'  or: 'http://saio:8080/v1/AUTH_account/c/o'
```

Comment:
Hi there. I just re-used this code as I need to generate temporary urls.
It took me a while to notice the issue but here it is in the doc "Do not URL-encode the path when you generate the HMAC-SHA1 signature. However, when you make the actual HTTP request, you should properly URL-encode the URL.".

I'd add this to the doc to ensure an non-encoded $path is passed in.